### PR TITLE
move connectToFlags to avoid circular dependency

### PR DESCRIPTION
**Pull request by glekner**

### CODE REVIEW

File: frontend/public/reducers/connectToFlags.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { FeatureState } from './features';
+
+export const stateToFlagsObject = (state: FeatureState, desiredFlags: string[]): FlagsObject =>
+  desiredFlags.reduce((allFlags, f) => ({ ...allFlags, [f]: state.get(f) }), {} as FlagsObject);
+
+const stateToProps = (state: FeatureState, desiredFlags: string[]): WithFlagsProps => ({
+  flags: stateToFlagsObject(state, desiredFlags),
+});
+
+export type FlagsObject = { [key: string]: boolean };
```
Comment (rawagner) [inline frontend/public/reducers/connectToFlags.ts:14]: this was moved but not deleted from  `reducers/features.ts ` together with some others

**Comment by christianvogt** [comment]
fwiw, I'm find with this change but didn't test run it

File: frontend/public/components/api-explorer.tsx [hunk 2]
```diff
@@ -26,7 +26,8 @@ import {
   ResourceAccessReviewRequest,
   ResourceAccessReviewResponse,
 } from '../module/k8s';
-import { connectToFlags, FlagsObject } from '../reducers/features';
+import { connectToFlags } from '@console/internal/reducers/connectToFlags';
```
Comment (vojtechszocs) [inline frontend/public/components/api-explorer.tsx:29]: `frontend/public` is `@console/internal` pkg so why not use relative path here?

File: frontend/public/components/app-contents.tsx [hunk 3]
```diff
@@ -8,7 +8,8 @@ import {
   useUserSettings,
   getPerspectiveVisitedKey,
 } from '@console/shared';
-import { connectToFlags, flagPending, FlagsObject } from '../reducers/features';
+import { connectToFlags } from '@console/internal/reducers/connectToFlags';
```
Comment (vojtechszocs) [inline frontend/public/components/app-contents.tsx:11]: Relative path?

File: frontend/public/components/chargeback.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import * as classNames from 'classnames';
 import { sortable } from '@patternfly/react-table';
 import { match as RMatch } from 'react-router-dom';
 
-import { connectToFlags } from '../reducers/features';
+import { connectToFlags } from '@console/internal/reducers/connectToFlags';
```
Comment (vojtechszocs) [inline frontend/public/components/chargeback.tsx:7]: Relative path?

File: frontend/public/components/routes/route-metrics.tsx [hunk 5]
```diff
@@ -2,7 +2,7 @@ import * as React from 'react';
 import { FLAGS } from '@console/shared/src/constants';
 import { Area } from '@console/internal/components/graphs/area';
 import { humanizeDecimalBytesPerSec } from '@console/internal/components/utils';
-import { connectToFlags, WithFlagsProps } from '../../reducers/features';
+import { connectToFlags, WithFlagsProps } from '../../reducers/connectToFlags';
 import { useTranslation } from 'react-i18next';
 import { K8sResourceKind } from '@console/internal/module/k8s';
```
Comment (vojtechszocs) [inline frontend/public/components/routes/route-metrics.tsx:7]: Relative paths instead of all `@console/internal` imports here?

**Review by vojtechszocs — CHANGES_REQUESTED** [review]

**Pull request merged**